packages/template-typescript-minimal/app/hooks/useProductSearch.ts

```diff
@@ -0,0 +1,43 @@
+import useSWR from 'swr'
+import {ShopperSearch} from 'commerce-sdk-isomorphic'
+import config from '../commerce-api-config'
+import {useServerEffect} from '../provider'
+import {getAccessToken} from '../auth'
+
+const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms))
+const isServer = typeof window === 'undefined'
+
+const getProductSearch = async (searchTerm) => {
+    const accessToken = await getAccessToken()
+
+    const api = new ShopperSearch({
+        ...config,
+        headers: {authorization: `Bearer ${accessToken}`},
+    })
+
+    await sleep(2000)
+
+    const result = await api.productSearch({
+        parameters: {q: searchTerm, limit: 20},
+    })
+
+    return result
+}
+
+const useProductSearch = ({searchTerm}, source: []) => {
+    if (isServer) {
+        const {data, isLoading, error} = useServerEffect(
+            async () => {
+                const data = await getProductSearch(searchTerm)
+                return data
+            },
+            source,
+            searchTerm
+        )
+        return {data, error, isLoading}
+    }
+    const {data, error, isValidating} = useSWR(searchTerm, getProductSearch)
```

Comment:
Initial examples on using `userSerferEffect` to create scapi hooks involve using it on both the client and the server. Will the initial execution of `useSWR` return data that was rendered on the server? Or are you populating SWR with the serialized data somewhere else ?

Reply:
In this POC, `useServerEffect` is used on server side only. and  `useSWR` is client side only.